test/e2e/instrumentation/events.go

```diff
@@ -0,0 +1,182 @@
+
+		ginkgo.By("getting the test event")
+		// get event by name
+		event, err := client.Get(context.TODO(), eventTestName, metav1.GetOptions{})
+		framework.ExpectNoError(err, "failed to fetch the test event")
+		framework.ExpectEqual(event.Note, eventPatchNote, "test event note does not match patch note")
+		framework.ExpectEqual(event.Reason, "Test", "test event reason should not be changed")
+		framework.ExpectEqual(event.Type, "Normal", "test event type should not be changed")
+		framework.ExpectEqual(event.ReportingController, "test-controller", "test event reportingController should not be changed")
+		framework.ExpectEqual(event.ReportingInstance, "test-node", "test event reportingInstance should not be changed")
+		framework.ExpectEqual(event.ObjectMeta.Name, eventTestName, "test event name should not be changed")
+
+		ginkgo.By("deleting the test event")
+		// delete original event
+		err = client.Delete(context.TODO(), eventTestName, metav1.DeleteOptions{})
+		framework.ExpectNoError(err, "failed to delete the test event")
+
+		ginkgo.By("listing all events in all namespaces")
+		// get a list of Events in all namespace
+		foundCreatedEvent = listEvents(eventTestName, f.Namespace.Name, clientAllNamespaces)
+		framework.ExpectEqual(foundCreatedEvent, false, "should not have found test event after deletion")
+
+		ginkgo.By("listing all events in framework namespace")
+		// get a list of Events in framework namespace
+		foundCreatedEvent = listEvents(eventTestName, f.Namespace.Name, client)
+		framework.ExpectEqual(foundCreatedEvent, false, "should not have found test event after deletion")
+	})
+
+	ginkgo.It("should delete a collection of events", func() {
+		eventTestNames := []string{"test-event-1", "test-event-2", "test-event-3"}
+
+		ginkgo.By("Create set of events")
+		// create a test event in test namespace
+		for _, eventTestName := range eventTestNames {
+			_, err := client.Create(context.TODO(), testEvent(eventTestName, "testevent-set"), metav1.CreateOptions{})
+			framework.ExpectNoError(err, "failed to create event")
+			framework.Logf("created %v", eventTestName)
+		}
+
+		ginkgo.By("get a list of Events with a label in the current namespace")
+		// get a list of events
+		eventList, err := client.List(context.TODO(), metav1.ListOptions{
+			LabelSelector: "testevent-set=true",
+		})
+		framework.ExpectNoError(err, "failed to get a list of events")
+		framework.ExpectEqual(len(eventList.Items), len(eventTestNames), "event list should have 3 items")
+
+		ginkgo.By("delete a list of events")
+		// delete a list of events
+		framework.Logf("requesting DeleteCollection of events")
+		err = client.DeleteCollection(context.TODO(), metav1.DeleteOptions{}, metav1.ListOptions{
+			LabelSelector: "testevent-set=true"})
+		framework.ExpectNoError(err, "failed to delete the test event")
+
+		ginkgo.By("check that the list of events matches the requested quantity")
+		pollFn := func() (bool, error) {
```

Comment:
Actually - you don't need to this. DeleteCollection is synchronous:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go#L1066

You can simply have:
eventList, err := client.List(...)
framework.ExpectNoError(err, "failed to list events")
framework.ExpectEqual(len(eventList.Items), 0, fmt.Sprintf("unexpected event list: %#v", eventList))

Reply:
Good to know. Thanks!